### PR TITLE
[exporter/datadog] Decouple `Config` structs from internal components

### PR DESCRIPTION
**Pull request by mx-psi**
**Description:** 

Decouple `config.Config` and related structs from internal components. In particular:

- Remove reference to `Config` on `GetHost`
- Remove reference to Config on ProcessMetrics
- Decouple metadata pusher configuration from main configuration

This is so that we can move config to the main package while avoiding cycles.

This is an internal refactor with no public changes.

**Link to tracking Issue:** #8373

**Testing:** 

- [x] amended unit tests.
- [x] end to end testing of host metadata.
- [x] check that the fallback hostname on the configuration is picked up.

### CODE REVIEW

**Review by KSerrania — APPROVED** [review]
The `PusherConfig` objects being abbreviated `mcfg` is slightly confusing, but otherwise LGTM.

**Comment by mx-psi** [comment]
> The `PusherConfig` objects being abbreviated `mcfg` is slightly confusing, but otherwise LGTM.

I changed that, originally I named it `MetadataPusherConfig`, but the linters don't like that 😄

**Comment by mx-psi** [comment]
Marking as `ready-to-merge` since it has a codeowner approval

**Review by jpkrohling — APPROVED** [review]

**Pull request merged**
